### PR TITLE
Publish a local release (canary or stable) to NPM

### PR DESCRIPTION
**Pull request by bvaughn**
#### Issue #14201. Builds on top of #14255 ([view the delta](https://github.com/bvaughn/react/compare/automated-release-scripts-part-3...bvaughn:automated-release-scripts-part-4?expand=1)).

This PR adds the final `publish` script and does a bit of cleanup of the pre-existing scripts. (For example, the `build` script is no longer needed and so it has been removed.)

This PR also removes the pre-build "version-check" behavior since there is an expected version mismatch between canary packages and the embedded React version (used by DevTools). If this change is controversial, I'm happy to elaborate on why I chose this detail and discuss alternatives.

**[Learn more about these new release scripts here.](https://github.com/bvaughn/react/blob/automated-release-scripts-part-4/scripts/release/README.md#react-release-scripts)**

### Checking out a Circle CI build for a canary release
prepare-canary

### Preparing a canary release to be promoted to stable
prepare-stable

### Publishing a release
publish-canary

### Resuming an interrupted publish (dry run)
publish-resume

### Creating a local canary (escape hatch)
Note that I trimmed the middle out of this GIF since building takes a long time.
create-canary-output

### CODE REVIEW

**Comment by bvaughn** [comment]
My last commit was totally unnecessary, but I thought I might as well replace our scattered `chalk` styles with a shared theme. This has the (admittedly small) benefit of making repeated things (e.g. package name or version) more recognizable at a glance, since they are consistently formatted now:

screen shot 2018-11-18 at 10 51 42 am
screen shot 2018-11-18 at 10 52 10 am
screen shot 2018-11-18 at 10 53 38 am
screen shot 2018-11-18 at 10 53 47 am
screen shot 2018-11-18 at 10 53 51 am
screen shot 2018-11-18 at 10 54 12 am
screen shot 2018-11-18 at 10 54 39 am
screen shot 2018-11-18 at 10 54 58 am
screen shot 2018-11-18 at 10 55 08 am

**Comment by bvaughn** [comment]
Rebased and added a new artifact to the Circle CI config– the error codes JSON. Going to let CI run to see if I configured it correctly. If so, I'm going to try to use this artifact when promoting a canary to stable so that I can avoid re-building it.

File: scripts/release/prepare-stable-commands/update-stable-version-numbers.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        writeFileSync(path, afterContents, {cwd});
+      }
+    });
+  }
+  writeFileSync(diffPath, diff, {cwd});
+  console.log(theme.header(`\n${numFilesModified} files have been updated.`));
+  console.log(
+    theme`A full diff is availbale at {path ${relative(cwd, diffPath)}}.`
+  );
+  await confirm('Do changes changes look correct?');
```
Comment (apalm) [inline scripts/release/prepare-stable-commands/update-stable-version-numbers.js]: Typo?

Comment (matpag) [reply scripts/release/prepare-stable-commands/update-stable-version-numbers.js]: Typo on "availbale" in line 161 too

Comment (bvaughn) [reply scripts/release/prepare-stable-commands/update-stable-version-numbers.js]: Whoops, yes.

**Comment by bvaughn** [comment]
Error codes are now stored as Circle CI artifacts and automatically downloaded (and staged) after publishing (for stable releases). If the published release was a local build, instructions are provided for updating error codes.

### CI build
screen shot 2018-11-19 at 12 58 02 pm

### Local build
screen shot 2018-11-19 at 1 01 11 pm

**Comment by bvaughn** [comment]
During our chat today, Sunil, it occurred to me that I could pretty easily auto-stage changes to package JSONs after a stable publish by just copying over portions of the just-published package JSON (4b99ae2).

screen shot 2018-11-20 at 3 04 42 pm

screen shot 2018-11-20 at 3 07 01 pm

**Comment by bvaughn** [comment]
🕺

**Comment by threepointone** [comment]
the code looks good to me, and we discussed this in person. I'm trying to actually run it on my machine today. 

ps - er, where do I approve this? I see this 
screenshot 2018-11-23 at 16 08 18

**Pull request merged**